src/openvic-simulation/country/Country.cpp

```diff
@@ -184,15 +170,10 @@ bool CountryManager::load_country_data_file(
 			),
 		"party", ZERO_OR_MORE, load_country_party(game_manager.get_politics_manager(), parties),
 		"unit_names", ZERO_OR_ONE,
-			game_manager.get_military_manager().get_unit_manager().expect_unit_dictionary(
-				[&unit_names, &name](Unit const& unit, ast::NodeCPtr value) -> bool {
-					if (unit_names.contains(&unit)) {
-						Logger::error("Country ", name, " has duplicate entry for ", unit.get_identifier(), " name list");
-						return false;
-					}
-					return name_list_callback([&unit_names, &unit](std::vector<std::string>&& list) -> bool {
-						return unit_names.emplace(&unit, std::move(list)).second;
-					})(value);
+			game_manager.get_military_manager().get_unit_manager().expect_unit_dictionary_reserve_length(
+				unit_names,
+				[&unit_names](Unit const& unit, ast::NodeCPtr value) -> bool {
+					return name_list_callback(map_callback(unit_names, &unit))(value);
```

Comment:
Unit names can be duplicate I believe. It makes them appear more often.

Reply:
`name_list_callback` parses the actual names and stores them in a vector of strings, the map here is from unit types to name lists, so each unit has a single list of names, but that list can include duplicate names.